src/nigui/private/windows/platform_impl.nim

```diff
@@ -1216,6 +1216,57 @@ method `enabled=`(button: NativeButton, enabled: bool) =
 
 
 # ----------------------------------------------------------------------------------------
+#                                      Checkbox
+# ----------------------------------------------------------------------------------------
+
+var pCheckboxOrigWndProc: pointer
+
+proc pCheckboxWndProc(hWnd: pointer, uMsg: int32, wParam, lParam: pointer): pointer {.cdecl.} =
+  case uMsg
+  of WM_KEYDOWN, WM_LBUTTONUP:
+    let checkbox = cast[Checkbox](pGetWindowLongPtr(hWnd, GWLP_USERDATA))
+    let keyCode = if uMsg == WM_KEYDOWN: cast[int](wParam) else: 32
+    if checkbox != nil and keyCode == 32:
+      checkbox.fChecked = not checkbox.fChecked
+      discard SendMessageA(checkbox.fHandle, BM_SETCHECK, cast[pointer](checkbox.fChecked), nil)
+
+      var event = new CheckboxToggleEvent
+      event.control = checkbox
```

Comment:
At this point the checkbox is cast to a generic `Control` which makes using `event.control` a bit messy, I think you have to recast it back to a `Checkbox`. Would like input on that